src/main/java/com/minecolonies/coremod/client/gui/WindowTownHall.java

```diff
@@ -931,6 +939,60 @@ private void createAndSetStatistics()
         findPaneOfTypeByID(COOKS_LABEL, Label.class).setLabelText(numberOfcooks);
     }
 
+    /**
+     * Creates several statistics and sets them in the townHall GUI.
+     */
+    private void createAndSetStatistics()
+    {
+        final int citizensSize = townHall.getColony().getCitizens().size();
+
+        Map<String, Integer> jobCountMap = new HashMap<String, Integer>();
+        for (@NotNull final CitizenDataView citizen : citizens)
+        {
+            int length = citizen.getJob().split("\\.").length;
+            String job = citizen.getJob().split("\\.")[length  - 1].toLowerCase();
+            System.out.println(job);
+            jobCountMap.put(job, jobCountMap.get(job) == null ? 1 : jobCountMap.get(job) + 1);
+        }
+        System.out.println(jobCountMap);
+
+        final String numberOfCitizens =
+                LanguageHandler.format("com.minecolonies.coremod.gui.townHall.population.totalCitizens", citizensSize, townHall.getColony().getMaxCitizens());
+
+        final DecimalFormat df = new DecimalFormat("#.#");
+        df.setRoundingMode(RoundingMode.CEILING);
+        final String roundedHappiness = df.format(building.getColony().getOverallHappiness());
+
+        findPaneOfTypeByID(HAPPINESS_LABEL, Label.class).setLabelText(roundedHappiness);
+        findPaneOfTypeByID(TOTAL_CITIZENS_LABEL, Label.class).setLabelText(numberOfCitizens);
+
+        Group group = findPaneOfTypeByID("citizen-stats", Group.class);
+        Integer unemployed = jobCountMap.get("") == null ? 0 : jobCountMap.get("");
+        jobCountMap.remove("");
+
+        final Label unemployedLabel = new Label();
+        unemployedLabel.setSize(100, 11);
+        unemployedLabel.setTextAlignment(Alignment.MIDDLE_LEFT);
+        unemployedLabel.setColor(BLACK, BLACK);
+        final String numberOfUnemployed = LanguageHandler.format("com.minecolonies.coremod.gui.townHall.population.unemployed", unemployed) ;
+        unemployedLabel.setLabelText(numberOfUnemployed);
+        group.addChild(unemployedLabel);
+
+        for (Map.Entry<String, Integer> entry : jobCountMap.entrySet())
+        {
+            final Label workerLabel = new Label();
+            workerLabel.setSize(100, 11);
+            workerLabel.setTextAlignment(Alignment.MIDDLE_LEFT);
+            workerLabel.setColor(BLACK, BLACK);
+            final String job = entry.getKey();
+            String labelJobKey = job.endsWith("man") ? job.replace("man", "men") : job + "s";
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Add parentheses to make the operator precedence explicit. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.minecolonies.com/coding_rules#rule_key=squid%3AS864)